src/CalculationClasses/MultiLayer/ReflectionMultiLayer.m

```diff
@@ -36,35 +35,34 @@ function Calculate(obj)
 
             obj.Setup_energy_mesh_full;
             obj.RecalculateToEnergyDistribution;
-            if obj.vacuum
-                R = zeros(size(obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:)));
-            else
-                R = obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:);
-            end
+            R = obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:); %bulk (semiinf) signal
             M = size(R,4)-1;
             N_E =size(R,3);
             wT = zeros(size(obj.FullEnergyDistribution{1}.T));
             Tw = zeros(size(obj.FullEnergyDistribution{1}.T));
             for i_layer = (obj.N_Layer-1):-1:1
-                Rlocal = obj.FullEnergyDistribution{i_layer}.R(:,:,:,:);
+                if obj.vacuum && i_layer == 1
+                    Rlocal = zeros(size(R));
+                else
+                    Rlocal = obj.FullEnergyDistribution{i_layer}.R(:,:,:,:);
+                end
                 if strcmp(obj.CalculationMethod{i_layer},'SLA')
                     wT = obj.FullEnergyDistribution{i_layer}.T;
                     Tw = obj.FullEnergyDistribution{i_layer}.T;
                 else
                     w = sparse(1:obj.ObjectsOfLayers{i_layer}.N, 1:obj.ObjectsOfLayers{i_layer}.N, obj.ObjectsOfLayers{i_layer}.mu_mesh_weights./obj.mu_mesh);
                     for m=0:M
                         for i=1:N_E
-                            if obj.vacuum && i_layer == obj.N_Layer-1
+                            if i_layer == 1 || i_layer == 2 % vacuum or surface

```

Comment:
Should we depend on layer number? I think no.

Reply:
Here I would eliminate it for the time being, because I have to think about this again. I will add it in the next PR